assessments/projects/contour/self-assessment.md

```diff
@@ -0,0 +1,208 @@
+# Contour Self-assessment
+
+## Table of contents
+
+* [Metadata](#metadata)
+  * [Security links](#security-links)
+* [Overview](#overview)
+  * [Actors](#actors)
+  * [Actions](#actions)
+  * [Background](#background)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|   |  |
+| -- | -- |
+| Software | https://github.com/projectcontour/contour |
+| Security Provider | No |
+| Languages | Go |
+| SBOM | TBD after discussion with project maintainers |
+| | |
+
+### Security links
+
+| Doc | url |
+| -- | -- |
+| Security file | https://github.com/projectcontour/contour/blob/main/SECURITY.md |
+| Default and optional configs | [Contour configuration](https://github.com/projectcontour/contour/blob/main/internal/contourconfig/contourconfiguration.go) <br /> https://projectcontour.io/docs/1.27/  |
+
+## Overview
+
+Contour is a Kubernetes ingress control providing the control plane for the Envoy edge and service proxy. Contour supports dynamic configuration updates and multi-team ingress-delegation out of the box, while maintaining a lightweight profile.
+
+### Background
+
+As an ingress controller, Contour manages incoming traffic for services hosted via Kubernetes clusters. Built on Envoy, an intermediary edge proxy between clients and services, Contour provides functionalities to route traffic to the appropriate services based on defined rules. 
+
+While Kubernetes already has base resources for ingress control, Contour utilizes those resources and extends Kubernetes’ capabilities for managing external access to services running within a cluster.
+
+### Actors
+There are three separate actors in this project’s system:
+- **_Kubernetes_**, which provides the clusters that Contour and Envoy deployments are made within
+- **_Contour_**, which acts as a management server for Envoy and provides it with configuration 
+- **_Envoy_**, which provides a high-performance reverse proxy.
+
+### Actions
+For Contour to serve as an ingress controller it must be deployed along with Envoy as separate containers in Kubernetes. Contour is a client of the Kubernetes API and watches it utilizing controller-runtime primitives. In the Envoy Pods, Contour runs as an init container and writes the configuration to a temporary volume. After initialization is complete, the Envoy container starts and retrieves the bootstrap configuration written by Contour.
+When a request is sent, first it is routed by a load-balancer to an instance of Envoy. The Envoy proxy then sends the request to a Contour pod.
```

Comment:
Something seems a bit fishy here... Considering that Kubernetes and Envoy are both external to Contour, it seems like they probably shouldn't be listed as "individual parts of your system that interact to provide the desired functionality."

Reply:
Referencing the image below, we aren't necessarily trying to understand the _entire image_. Instead, we want to put a magnifying glass over _contour_ and look at what actors and actions are interacting within the context of that service.

![image](https://github.com/cncf/tag-security/assets/21176439/1e86b012-89a0-4a23-b998-3c373434da24)

Reply:
Thanks for the explanation! With that in mind, we've updated the actors and actions to better reflect the Contour controller itself, rather than just its external actors.